app/src/components/panels/SupportProposal.js

```diff
@@ -0,0 +1,197 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import BN from 'bn.js'
+import { useAragonApi } from '@aragon/api-react'
+import {
+  Button,
+  ButtonBase,
+  Field,
+  GU,
+  Info,
+  TextInput,
+  useSidePanelFocusOnReady,
+  useTheme,
+} from '@aragon/ui'
+import { toDecimals, round, pct } from '../../lib/math-utils'
+import useAccountTotalStaked from '../../hooks/useAccountTotalStaked'
+import { formatTokenAmount } from '../../lib/token-utils'
+
+const SupportProposal = React.memo(function SupportProposal({ id, onDone }) {
+  const theme = useTheme()
+  const [amount, setAmount] = useState({
+    value: '0',
+    valueBN: new BN(0),
+  })
+
+  const {
+    api,
+    appState: { stakeToken },
+  } = useAragonApi()
+  const inputRef = useSidePanelFocusOnReady()
+
+  const totalStaked = useAccountTotalStaked()
+  const nonStakedTokens = stakeToken.balanceBN.sub(totalStaked)
+
+  const handleEditMode = useCallback(
+    editMode => {
+      setAmount(amount => ({
+        ...amount,
+        value: amount.valueBN.gte(0)
+          ? formatTokenAmount(
+              amount.valueBN,
+              stakeToken.tokenDecimals,
+              false,
+              false,
+              {
+                commas: !editMode,
+                replaceZeroBy: editMode ? '' : '0',
+                rounding: stakeToken.tokenDecimals,
+              }
+            )
+          : '',
+      }))
+    },
+    [stakeToken]
+  )
+
+  // Amount change handler
+  const handleAmountChange = useCallback(
+    event => {
+      const newAmount = event.target.value
+
+      const newAmountBN = new BN(
+        isNaN(event.target.value)
```

Comment:
in which case the target.value isNaN?

Reply:
When the user inputs something rather than numbers or an invalid number format (e.g `dafsdf` or `12.3.43.5`)